dimod/binary_quadratic_model.py

```diff
@@ -1953,6 +1955,134 @@ def from_numpy_matrix(cls, mat, variable_order=None, offset=0.0, interactions=No
 
         return bqm
 
+    def to_numpy_vectors(self, variable_order=None, dtype=None, index_dtype=None, sort_indices=False):
+        """Convert a binary quadratic model to numpy arrays.
+
+        Args:
+            variable_order (iterable, optional):
+                If provided, labels the variables; otherwise, row/column indices are used.
+
+            dtype (:class:`numpy.dtype`, optional):
+                Data-type of the biases. By default, the data-type is inferred from the biases.
+
+            index_dtype (:class:`numpy.dtype`, optional):
+                Data-type of the indices. By default, the data-type is inferred from the labels.
+
+            sort_indices (bool, optional, default=False):
```

Comment:
It might be good to split this into two separate options. Sorting column-wise costs almost nothing, but sorting row-wise in a large dense problem can be expensive.

Reply:
Hmm, I think in this case I am inclined to let the user sort it out themselves.